packages/block-editor/src/hooks/behaviors.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+/**
+ * TODO: Add description.
+ *
+ * @param {WPComponent} BlockEdit Original component.
+ *
+ * @return {WPComponent} Wrapped component.
+ */
+export const withBehaviors = createHigherOrderComponent( ( BlockEdit ) => {
+	return ( props ) => {
+		const { behaviors: blockBehaviors } = props.attributes;
+		const settings = select( blockEditorStore ).getSettings();
+		const themeBehaviors =
+			settings?.__experimentalFeatures?.blocks?.[ 'core/image' ]
+				.behaviors;
```

Comment:
If anyone knows of a more proper way to access the block-level settings from the
`theme.json`, I'm all ears!

Reply:
Not more idiomatic, but I guess using props.name instead of the core block hard code name would be more scalable. Something like:

```
const themeBehaviors = settings?.__experimentalFeatures?.blocks?.[ props.name ]?.behaviors;
```

Reply:
Makes sense 🙂 

Even though, for now, we're only enabling the Behaviors on the Image block, we should still change it so that we don't have to update this line in the future.

Reply:
Done dd70de9b14337e164dc23f5b076889dbf468c5fd